src/NuGet.Services.AzureSearch/DocumentUtilities.cs

```diff
@@ -0,0 +1,239 @@
+        /// Example A (no target frameworks):
+        ///     Microsoft.Data.OData:5.0.2|Microsoft.WindowsAzure.ConfigurationManager:1.8.0
+        /// Example B (target frameworks):
+        ///     NETStandard.Library:1.6.1:netstandard1.0|Newtonsoft.Json:10.0.2:netstandard1.0
+        /// Example C (just target framework):
+        ///     ::net20|::net35|::net40|::net45|NETStandard.Library:1.6.1:netstandard1.0
+        /// </summary>
+        private static string GetFlattenedDependencies(PackageDetailsCatalogLeaf leaf)
+        {
+            if (leaf.DependencyGroups == null)
+            {
+                return null;
+            }
+
+            var builder = new StringBuilder();
+            foreach (var dependencyGroup in leaf.DependencyGroups)
+            {
+                var targetFramework = dependencyGroup.ParseTargetFramework();
+
+                if (dependencyGroup.Dependencies != null && dependencyGroup.Dependencies.Any())
+                {
+                    foreach (var packageDependency in dependencyGroup.Dependencies)
+                    {
+                        AddFlattennedPackageDependency(targetFramework, packageDependency, builder);
+                    }
+                }
+                else
+                {
+                    if (builder.Length > 0)
+                    {
+                        builder.Append("|");
+                    }
+
+                    builder.Append(":");
+                    AddFlattenedFrameworkDependency(targetFramework, builder);
+                }
+            }
+
+            return builder.Length > 0 ? builder.ToString() : null;
+        }
+
+        private static void AddFlattennedPackageDependency(
+            NuGetFramework targetFramework,
+            PackageDependency packageDependency,
+            StringBuilder builder)
+        {
+            if (builder.Length > 0)
+            {
+                builder.Append("|");
+            }
+
+            builder.Append(packageDependency.Id);
+            builder.Append(":");
+
+            var versionRange = packageDependency.ParseRange();
+
+            if (!VersionRange.All.Equals(versionRange))
+            {
+                builder.Append(versionRange?.ToString("S", new VersionRangeFormatter()));
+            }
+
+            AddFlattenedFrameworkDependency(targetFramework, builder);
+        }
+
+        private static void AddFlattenedFrameworkDependency(NuGetFramework targetFramework, StringBuilder builder)
+        {
+            if (!SpecialFrameworks.Contains(targetFramework))
+            {
+                try
+                {
+                    builder.Append(":");
+                    builder.Append(targetFramework?.GetShortFolderName());
+                }
+                catch (FrameworkException)
+                {
+                    // ignoring FrameworkException on purpose - we don't want the job crashing
+                    // whenever someone uploads an unsupported framework
```

Comment:
This could result in the following output string:

    ::

Is that valid or even useful?

Reply:
Valid? Yes. This is existing behavior. Useful? Well, this is essentially the same assertion as a `<group />` element in the .nuspec... which today is noise (all other dependency groups are more applicable than the "any" TFM. But again this is existing behavior so it's not worth the risk to change it.